src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java

```diff
@@ -235,8 +235,12 @@ protected final void verifyWithInlineConfigParser(String filePath, String... exp
                 InlineConfigParser.parse(filePath);
         final DefaultConfiguration parsedConfig =
                 testInputConfiguration.createConfiguration();
-        verifyViolations(parsedConfig, filePath, testInputConfiguration.getViolations());
-        verify(parsedConfig, filePath, expected);
+        final List<String> actualViolations = getActualViolationsForFile(parsedConfig, filePath);
+        verifyViolations(filePath, testInputConfiguration.getViolations(), actualViolations);
+        
+        assertWithMessage("Violations for %s differ.", filePath)
+            .that(actualViolations)
+            .containsExactlyElementsIn(expected);
```

Comment:
Instead of running a check twice. once to verify the violation in `verifyViolations()` against testInputConfigurationViolation that got parsed from the input file. and another one with `verify()` to verify violations against the expected array.

I extracted running the check out  to `getActualViolationsForFile()` once and then used these actualViolations to verify them against the  testInputConfigurationViolation with `verifyViolations()` and then verify them with the expected array with  assertion